### ISSUE TITLE
[Feature Request]: All black menu bar background to hide notch

### ISSUE THREAD

**Issue by svoop** [jordanbaird/Ice#82]
### Description

The "tint" setting for the menu bar appearance doesn't allow for the menu bar background to be solid black (and therefore hide the notch). 

This is how things look with one of the default color wallpapers:

![Screenshot 2024-06-05 at 19 59 02](https://github.com/jordanbaird/Ice/assets/7255/a0ad988f-57fd-41ca-926d-b22cfd157a4a)

And after adding a solid black tint with Ice:

![Screenshot 2024-06-05 at 19 59 14](https://github.com/jordanbaird/Ice/assets/7255/ff790ba5-4307-42de-9fb1-09d11e963f6d)

Would it be possible/complicated to add the ability to set the background to a solid black like so:

![Screenshot 2024-06-05 at 19 59 35](https://github.com/jordanbaird/Ice/assets/7255/2f4cb4c8-0ae0-4af3-9863-b7079f7fe586)

Notch gone :smile:

This would make other tools for just that purpose (e.g. Top Notch or ChangeMenuBarColor) obsolete.

Thanks for considering this... and for Ice! (Just switched and donated to Ice due to the [ongoing Bartender concerns](https://www.reddit.com/r/macapps/comments/1d7zjv8/comment/l76r88g/).)

**Comment by jordanbaird** [comment]
Yeah, we could do this. I had previously kind of decided that it wasn't within scope for Ice, but there have been a decent number of requests for this feature, and thinking about it more, it probably actually is within scope.

**Comment by leoravagnani** [comment]
@jordanbaird doing this via Ice would be the way to go!

Top Notch watches the current wallpaper and then generate a version of the chosen image with a black bar and then set the wallpaper to this new image under the hood. It would not behave correctly for me sometimes when using dynamic wallpapers and just stopped working at all when upgrading to Sonoma and I ditched it. Now I just stick to a mostly black wallpaper to get the same effect but would be nice to have the freedom to change it and still maintain the full black bar.

**Comment by jordanbaird** [comment]
@leoravagnani Yeah, dynamic wallpapers not working is a limitation of this feature. Unfortunately, changing the wallpaper is the only way macOS will allow the menu bar color to change. But as long as people understand that limitation, and are okay with it, I'm willing to implement it.

**Comment by leoravagnani** [comment]
So the tint feature and styling feature is actually changing the wallpaper then? Or these are different scenarios?

**Comment by jordanbaird** [comment]
@leoravagnani It's actually overlaying a partially transparent window above the menu bar. There are a lot of things that can be done to the appearance and behavior of a window to remove all the peripheral elements (like the close, minimize, and zoom buttons, borders, title bars, etc.). In fact, that's what the menu bar itself is. So you can think of it as adding a Photoshop layer over the menu bar.

**Comment by leoravagnani** [comment]
@jordanbaird hmmm got it! so that's why adding a tint makes the icons a bit darker then, and why a 100% opaque tint would not work as it would effectively hide everything beneath. So the only way around the hiding the notch would be changing the wallpaper.

**Comment by Xytronix** [comment]
> @leoravagnani Yeah, dynamic wallpapers not working is a limitation of this feature. Unfortunately, changing the wallpaper is the only way macOS will allow the menu bar color to change. But as long as people understand that limitation, and are okay with it, I'm willing to implement it.

I would add a disclaimer or a note that this is an experimental feature and doesn’t work with dynamic wallpapers. You can add blur, color (any color (HEX field, color selection or black/white)

**Comment by svoop** [comment]
@leoravagnani wrote:

> Top Notch watches the current wallpaper and then generate a version of the chosen image with a black bar and then set the wallpaper to this new image under the hood.

Actually, it does a very bad job at this and while the resulting wallpaper does have the solid black menu bar background, the actual wallpaper gets distorted.

The disclaimer note proposed by @Xytronix should do the trick IMO.

**Comment by Xytronix** [comment]
![Untitled](https://github.com/user-attachments/assets/cd39a87d-78af-4dcb-aabc-604f38f9a5a4)

**Comment by ryanwwest** [comment]
I was writing a FR to add opacity control to the Menu Bar Appearance 'tint' (e.g. set opacity to 100% for a totally solid-colored top bar, not necessarily just black) but then found this. Top Notch stopped working for me too, so this would entirely replace that app _and_ make the menu bar items more pleasing.

**Comment by beratbayram** [comment]
I would love to have this feature, and we don't need anything advanced; just a pure black top bar would be sufficient.

**Comment by hongyinull** [comment]
Is it possible for this feature to be developed? I came here for the same reason and willing to donate and support, just because the top-notch one really isn't working XD.

**Comment by hiltbru** [comment]
Also commenting to say that this is easily the most anticipated feature of Ice for me. Would really love to have this option, especially used in conjunction with the "dynamic appearance" option.

**Comment by goshatch** [comment]
Thank you for Ice! I'd love to have this feature, as well.

**Comment by Albusquerque** [comment]
+1

**Comment by Sebeb** [comment]
So I had a go at quickly vibe-coding this feature in. It's definitely very rough around the edges (pun not intended), hence the draft status of the PR, see its description for further details. But it does mostly work, I've been using it for a while, and it's quite nice. I also added optionally rounded corner radii beneath the status bar and at the screen's bottom, to mirror the MacBook's rounded corners. Since it's an overlay layer, and not changing the wallpaper like TopNotch, it works in fullscreen apps, and also works with video wallpaper apps like Wallspace.

<img width="3456" height="2234" alt="Screenshot 2026-07-07 at 11 36 24" src="https://github.com/user-attachments/assets/0a7d555c-6950-472e-8edb-1151fbcd21b9" />
<img width="1814" height="1426" alt="Screenshot 2026-07-07 at 11-44-20 —" src="https://github.com/user-attachments/assets/cb8d34ce-a1bf-41df-8cb8-ae56ad7841d9" />

Here's the draft PR. If anyone wants to have a go at tidying it up (or just starting from scratch and implementing this properly), be my guest !

https://github.com/jordanbaird/Ice/pull/960

If you just want to try the feature yourself, you can download it here: https://github.com/Sebeb/Ice/releases/tag/menu-bar-backdrop-screen-corners-draft

**Comment by AlecRust** [comment]
Open the PR on the maintained project Thaw
